amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/GracefulShutdownCoordinator.java

```diff
@@ -62,6 +67,21 @@ private String awaitingFinalShutdownMessage(GracefulShutdownContext context) {
             return String.format("Waiting for %d record processors to complete final shutdown", outstanding);
         }
 
+        /**
+         * used to wait for the worker's final shutdown to complete before returning the future for graceful shutdown
+         * @return true if the final shutdown is successful, false otherwise.
+         */
+        private boolean waitForFinalShutdown(GracefulShutdownContext context) {
```

Comment:
Is there any particular reason why we would want to catch the `InterruptedException` within the try-catch block of the `call` method? IMO it makes sense to catch it within `waitForFinalShutdown` and return false since the exception means that the graceful shutdown didn't complete correctly.

Reply:
> return false since the exception means that the graceful shutdown didn't complete correctly.

What's the benefit for returning false instead of just allowing the exception to bubble up? Either way they mean that the shutdown didn't complete correctly right? I don't see we are doing extra handling for false shutdowns and we are not reseting the interrupt flag `Thread.currentThread().interrupt()` not sure if that's a good practice